doc: nrf: bluetooth: RAS RREQ and RRSP docs (#19230)
nrfconnect/sdk-nrf: doc/nrf/libraries/bluetooth/services/rreq.rst

.. _rreq_readme:

Ranging Requestor (RREQ)
########################

.. contents::
   :local:
   :depth: 2

Overview
********

This library implements the Ranging Requestor for Channel Sounding with the corresponding set of characteristics defined in the `Ranging Service Specification`_ and the `Ranging Profile Specification`_.

This library supports On Demand Ranging Data.

Configuration
*************

> To enable this library, use the :kconfig:option:`CONFIG_BT_RAS` Kconfig option.

Reviewer: ```suggestion
To enable this library, use the :kconfig:option:`CONFIG_BT_RAS` and :kconfig:option:`CONFIG_BT_RAS_RREQ` Kconfig options.
```
Maybe this makes more sense

.. _rreq_readme:

Ranging Requestor (RREQ)
########################

.. contents::
   :local:
   :depth: 2

Overview
********

This library implements the Ranging Requestor for Channel Sounding with the corresponding set of characteristics defined in the `Ranging Service Specification`_ and the `Ranging Profile Specification`_.

This library supports On Demand Ranging Data.

Configuration
*************

* :kconfig:option:`CONFIG_BT_RAS` - Enables RREQ and RRSP Kconfig options.

* :kconfig:option:`CONFIG_BT_RAS_MAX_ANTENNA_PATHS` - Sets the maximum number of antenna paths supported by the device.
  This sets the antenna paths for each step that can be stored inside the Ranging Service.
  This value must match the supported Channel Sounding capabilities of the device.

>   This affects the per-instance memory usage of the Ranging Service.

Reviewer: This doesn't really apply for rreq. Would maybe just keep the first line

* :kconfig:option:`CONFIG_BT_RAS_MODE_3_SUPPORTED` - Sets support for storing Mode 3 Channel Sounding steps.
  This will allocate memory for the Ranging Service to store Mode 3 Channel Sounding steps.
  This value must match the supported Channel Sounding capabilities of the device.
  This affects the per-instance memory usage of the Ranging Service.

* :kconfig:option:`CONFIG_BT_RAS_RREQ` - Enables RREQ Kconfig options.

* :kconfig:option:`CONFIG_BT_RAS_RREQ_MAX_ACTIVE_CONN` - Sets the number of simultaneously supported RREQ instances.

* :kconfig:option:`CONFIG_HEAP_MEM_POOL_ADD_SIZE_RAS_RREQ_GATT_DM` - Sets the heap size for the GATT distance measurement module.

* :kconfig:option:`CONFIG_BT_RAS_RREQ_LOG_LEVEL` - Sets the logging level of the RREQ library.

Usage
*****

You can set up the RREQ either as a Channel Sounding Initiator or Reflector.

| See the sample: :file:`samples/bluetooth/channel_sounding_ras_initiator`

> | See the sample: :file:`samples/bluetooth/channel_sounding_ras_reflector`

Reviewer: ```suggestion
```

Only top sample is valid

.. _rreq_readme:

Ranging Requestor (RREQ)
########################

.. contents::
   :local:
   :depth: 2

Overview
********

This library implements the Ranging Requestor for Channel Sounding with the corresponding set of characteristics defined in the `Ranging Service Specification`_ and the `Ranging Profile Specification`_.

Supported features
==================

> - On Demand Ranging Data

Reviewer 2: ```suggestion
The library supports On Demand Ranging Data.
```
The Supported features heading could be left out.

Configuration
*************

> Enable RREQ and RRSP kconfig options using the :kconfig:option:`CONFIG_BT_RAS` kconfig option.

Reviewer 2: ```suggestion
To enable the RREQ and RRSP Kconfig options, use :kconfig:option:`CONFIG_BT_RAS`.
```

> Set the maximum number of antenna paths supported by the device using the :kconfig:option:`CONFIG_BT_RAS_MAX_ANTENNA_PATHS` kconfig option.

Reviewer 2: ```suggestion
Set the maximum number of antenna paths supported by the device using the :kconfig:option:`CONFIG_BT_RAS_MAX_ANTENNA_PATHS` Kconfig option.
```

> This sets the antenna paths per step that can be stored inside the Ranging Service.

Reviewer 2: ```suggestion
This sets the antenna paths for each step that can be stored inside the Ranging Service.
```

This value must match the supported Channel Sounding capabilities of the device.
This affects the per-instance memory usage of the Ranging Service.

> Set support for storing Mode 3 Channel Sounding steps using the :kconfig:option:`CONFIG_BT_RAS_MODE_3_SUPPORTED` kconfig option.

Reviewer 2: ```suggestion
To set support for storing Mode 3 Channel Sounding steps, use the :kconfig:option:`CONFIG_BT_RAS_MODE_3_SUPPORTED` Kconfig option.
```

This will allocate memory for the Ranging Service to store Mode 3 Channel Sounding steps.
This value must match the supported Channel Sounding capabilities of the device.
This affects the per-instance memory usage of the Ranging Service.

> Enable RREQ kconfig options using the :kconfig:option:`CONFIG_BT_RAS_RREQ` kconfig option.

Reviewer 2: ```suggestion
To enable the RREQ Kconfig options, use :kconfig:option:`CONFIG_BT_RAS_RREQ`.
```

> Set the number of simultaneously supported RREQ instances using the :kconfig:option:`CONFIG_BT_RAS_RREQ_MAX_ACTIVE_CONN` kconfig option.

Reviewer 2: ```suggestion
To set the number of simultaneously supported RREQ instances, use the :kconfig:option:`CONFIG_BT_RAS_RREQ_MAX_ACTIVE_CONN` Kconfig option.
```

> Set the heap size for the GATT distance measurement module using the :kconfig:option:`CONFIG_HEAP_MEM_POOL_ADD_SIZE_RAS_RREQ_GATT_DM` kconfig option.
> 
> Set the logging level of the RREQ library using the :kconfig:option:`CONFIG_BT_RAS_RREQ_LOG_LEVEL` Kconfig option.

Reviewer 2: ```suggestion
To set the heap size for the GATT distance measurement module, use the :kconfig:option:`CONFIG_HEAP_MEM_POOL_ADD_SIZE_RAS_RREQ_GATT_DM` Kconfig option.

To set the logging level of the RREQ library, use the :kconfig:option:`CONFIG_BT_RAS_RREQ_LOG_LEVEL` Kconfig option.
```

Usage
*****

> The RREQ can be setup as either a Channel Sounding Initiator or Reflector.

Reviewer 2: ```suggestion
You can set up the RREQ either as a Channel Sounding Initiator or Reflector.
```
